pkg/components/frontend.go

```diff
@@ -141,10 +142,29 @@ func (f *frontend) IsRunning(_ context.Context) bool {
 	return true
 }
 
-func generateAddrArg(addr string, nodeId int) string {
+// formatAddrArg formats the given addr and nodeId to a valid socket string.
+// This function will return an empty string when the given addr is empty.
+func formatAddrArg(addr string, nodeId int) string {
+	// return empty result if the address is not specified
+	if len(addr) == 0 {
```

Comment:
Will GreptimeDB use a default value if this arg is empty?

Reply:
All three components have a default value. And `gtctl` will also pass a config file which has related options.

Reply:
Got. Than we may also return `""` for  `generateMetaSrvAddr` and `generateDatanodeAddr ` if `addr` is empty.

Reply:
👍 I've refactored them to use the same function in https://github.com/GreptimeTeam/gtctl/pull/195/commits/cc591fa6c0e5e8abbb6a5977ebe6dfb8f5810cb8